src/Templating/MutableTemplateRegistry.php

```diff
@@ -22,7 +22,7 @@ final class MutableTemplateRegistry extends AbstractTemplateRegistry implements
     // public function setTemplates(array $templates): void
     public function setTemplates(array $templates)
     {
-        $this->templates = $templates;
+        $this->templates = array_merge($this->templates, $templates);
```

Comment:
Is this change needed ? Seems like a BC-break and change a lot this method.

According to https://symfony.com/doc/current/bundles/SonataAdminBundle/reference/templates.html
```
A setTemplates(array $templates) (notice the plural) method also exists, that allows you to set multiple templates at once. Notice that, if used outside of the service definition context, setTemplates(array $templates) will replace the whole template list for that Admin class, meaning you have to explicitly pass the full template list in the $templates argument.
```

So it seems expected to override the whole template list to me.

There is already 
```
if ('setTemplates' === $method) {
                $definedTemplates = array_merge($definedTemplates, $args[0]);

                continue;
            }
```
In the AddDependencyCallsCompilerPass

Reply:
We can change methods for `overrideTemplate()` and `overrideTemplates()` but it is bad idea.

When we call method `setTemplates()` in service config, we will use `array_merge`.
When someone call `AbstractAdmin::setTemplates()` in code then templates will be replace.

The same method work in two diffrence way.

We should fix it. If we cant do it now, it should be mark as `NEXT_MAJOR`.

Reply:
I don't think someone will use it to override every template at once.
So I think the array_merge is not a bad idea.

But since it's a BC-break, I think the `NEXT_MAJOR: use array merge and update the doc https://github.com/sonata-project/SonataAdminBundle/blob/3.x/docs/reference/templates.rst` is better indeed.

Reply:
Done.